enhancements/insights/config-api.md

```diff
@@ -0,0 +1,288 @@
+	// Enables user to force Insights data gathering by setting a new reason.
+	// When there is some gathering in the progress then it is interrupted.
+	// When all the gatherers are deactivated by the `DisabledGatherers`, nothing happens.
+	ForceGatherReason string `json:"forceGatherReason,omitempty"`
+	// List of gatherers to be excluded from the gathering. All the gatherers can disabled by providing "all" value.
+	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	// The particular gatherers IDs can be found at https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md.
+	// An example of disabling some gatherers can look like `disabledGatherers: ["clusterconfig/machine_configs", "workloads/workload_info"]`
+	DisabledGatherers []string `json:"disabledGatherers,omitempty"`
+}
+
+type GatheringStatus struct {
+	// LastGatherTime is the last time when Insights gathering finished.
+	LastGatherTime metav1.Time `json:"lastGatheringTime,omitempty"`
+	// Reason provides last known reason of gathering. This is helpful
+	// especially when gathering was forced by user
+	Reason string `json:"reason,omitempty"`
+	// Status is waiting or running. If it is running then it means there is some gathering in progress.
+	// If it is waiting then a new gathering can be triggered immediately.
+	Status GatherStatus `json:"status,omitempty"`
+	// List of active gatherers (and their statuses) in the last gathering.
+	GathererStatuses []GathererStatus `json:"gathererStatuses,omitempty"`
+}
+
+type ReportStatus struct {
+	// Number of active Insights healthchecks with low severity
+	LowCount int `json:"low,omitempty"`
+	// Number of active Insights healthchecks with moderate severity
+	ModerateCount int `json:"moderate,omitempty"`
+	// Number of active Insights healthchecks with important severity
+	ImportantCount int `json:"important,omitempty"`
+	// Number of active Insights healthchecks with critical severity
+	CriticalCount int `json:"critical,omitempty"`
+	// TotalCount is the count of all active Insights healthchecks
+	TotalCount int `json:"total,omitempty"`
+}
+
+type GathererStatus struct {
+	// Name is the name of the gatherer.
+	Name string `json:"name"`
+	// GathererConditions provide details on the status of each gatherer.
+	GathererConditions []GathererCondition `json:"conditions"`
+	// DurationMillisecond represents the time spent gathering.
+	DurationMillisecond int64 `json:"durationMillisecond"`
+	// RecordsCount represents the number of records produced by the gatherer.
+	RecordsCount int `json:"recordsCount"`
+}
+
+type GathererCondition struct {
+	// Type of the gatherer condition
+	Type GathererConditionType `json:"type"`
+	// Status is last known status of the particular gatherer condition
+	Status corev1.ConditionStatus `json:"status"`
+	// Messages is an optional attribute that provides error and warning messages
+	// from the gatherer
+	Messages []string `json:"messages,omitempty"`
```

Comment:
we usually just join all messages into one big string... i guess you rarely need this as slice.

Reply:
Newly discussed in the https://github.com/openshift/enhancements/pull/1037/files#r856243714